cmd/cloud-endpoints-controller/main.go

```diff
@@ -364,7 +392,34 @@ func calcParentSig(parent *CloudEndpoint, addStr string) string {
 	return fmt.Sprintf("%x", hasher.Sum(nil))
 }
 
-func makeDefaultOpenAPISpec(endpoint, target string, jwtAudiences []string) (string, error) {
+func validateOpenAPISpec(specOriginal string) error {
+	var spec map[string]interface{}
+	return yaml.Unmarshal([]byte(specOriginal), &spec)
+}
+
+func getUserSuppliedOpenAPISpec(parent *CloudEndpoint, status *CloudEndpointControllerStatus) (string, error) {
+	var specString string
+	var err error
+	if parent.Spec.OpenAPISpec != nil {
+		specYaml, err := yaml.Marshal(parent.Spec.OpenAPISpec)
+		if err != nil {
+			log.Printf("[ERROR][%s] Supplied spec: %s", parent.Name, err.Error())
```

Comment:
Consider just using `%v, err` to get the error message rather than `err.Error()`, that's how it's done everywhere else in the code.